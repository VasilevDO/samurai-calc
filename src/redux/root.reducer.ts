import {combineReducers} from 'redux';
import SamuraiCalcSlice from './samuraiCalc/SamuraiCalc.slice';

const rootReducer = combineReducers({
	samuraiCalc: SamuraiCalcSlice.reducer,
});

export default rootReducer;
