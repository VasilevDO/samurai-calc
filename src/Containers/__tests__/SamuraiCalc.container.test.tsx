import {fireEvent, render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import SamuraiCalc from '../SamuraiCalc.container';
import * as reduxHooks from '../../hooks/redux.hook';

const useDispatchSpy = jest.spyOn(reduxHooks, 'useAppDispatch');
const mockDispatchFn = jest.fn();
useDispatchSpy.mockReturnValue(mockDispatchFn);

const useSelectorSpy = jest.spyOn(reduxHooks, 'useAppSelector');
const initialState = {
	screen: '123', history: ['1', '2', '3'], historyLength: 3, isTouchScreen: false,
};
useSelectorSpy.mockReturnValue(initialState);

describe('With React Testing Library', () => {
	test('should render button with correct text (no style provided)', () => {
		const {container} = render(<SamuraiCalc />);
		expect(container).toMatchSnapshot();
	});
});
