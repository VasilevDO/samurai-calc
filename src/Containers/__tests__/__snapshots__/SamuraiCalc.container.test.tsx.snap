// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library should render button with correct text (no style provided) 1`] = `
.c5 {
  display: table-cell;
  vertical-align: middle;
  height: calc(2em + 8px);
  width: calc(2em + 8px);
  margin: 0 10px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  font-weight: inherit;
  text-align: center;
  border: none;
  color: white;
  background-color: inherit;
}

.c5:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c5:hover {
  background-color: rgba(255,255,255,0.12);
}

.c6 {
  display: table-cell;
  vertical-align: middle;
  height: calc(2em + 8px);
  width: calc(2em + 8px);
  margin: 0 10px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  font-weight: inherit;
  text-align: center;
  border: none;
  color: #2B589A;
  background-color: #FFFFFF;
}

.c6:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c6:hover {
  background-color: rgba(255,255,255,0.7);
}

.c4 {
  display: grid;
  grid-template-columns: repeat(4,auto);
  grid-gap: 16px 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-flex-flow: column-reverse;
  -ms-flex-flow: column-reverse;
  flex-flow: column-reverse;
  height: 117px;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  overflow: hidden;
}

.c2 > p {
  text-align: right;
  overflow-wrap: anywhere;
}

.c3 {
  width: 100%;
  height: 76px;
  color: inherit;
  text-align: right;
  background-color: inherit;
  border: none;
  margin-bottom: 18px;
  font-weight: 700;
  font-size: 56px;
  line-height: 80px;
}

.c3:focus {
  outline: none;
}

.c0 {
  display: inline-block;
  padding: 38px 31px;
  background: #FFFFFF33;
  -webkit-backdrop-filter: blur(4px);
  backdrop-filter: blur(4px);
  border-radius: 18px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Geometria';
  font-style: normal;
  font-weight: 500;
  font-size: 36px;
  line-height: 88px;
  text-align: center;
  color: #F2F2F2;
  padding: 16px 48px 44px 46px;
  border-radius: 18px;
  background: linear-gradient(155.23deg,#28518E 0%,#3A77D1 100%);
  box-shadow: 0px 82px 158px rgba(0,0,0,0.35),0px 24.7206px 47.6324px rgba(0,0,0,0.228056),0px 10.2677px 19.7841px rgba(0,0,0,0.175),0px 3.71362px 7.1555px rgba(0,0,0,0.121944);
}

.c1 > hr {
  width: 100%;
  border: 1px solid #FFFFFF;
  opacity: 0.35;
  margin-bottom: 37px;
}

.c1 > * {
  max-width: 460px;
}

.c1 > *:first-child {
  margin-bottom: 28px;
}

.c1 > *:nth-child(2) {
  margin-bottom: 16px;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    height: 1.5em;
    font-weight: 500;
    font-size: 28px;
    line-height: 40px;
  }

  .c3 >:not(:last-child) {
    margin-bottom: 14px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 10px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 20px;
    line-height: 36px;
    padding: 20px;
  }

  .c1 > *:not(:last-child) {
    margin-bottom: 16px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="history"
      >
        <p>
          3
        </p>
        <p>
          2
        </p>
        <p>
          1
        </p>
      </div>
      <input
        class="c3"
        data-testid="input"
        value="123"
      />
      <hr />
      <div
        class="c4"
        data-testid="controls"
      >
        <button
          class="c5"
        >
          C
        </button>
        <button
          class="c5"
        >
          √
        </button>
        <button
          class="c5"
        >
          %
        </button>
        <button
          class="c5"
        >
          /
        </button>
        <button
          class="c5"
        >
          7
        </button>
        <button
          class="c5"
        >
          8
        </button>
        <button
          class="c5"
        >
          9
        </button>
        <button
          class="c5"
        >
          ×
        </button>
        <button
          class="c5"
        >
          4
        </button>
        <button
          class="c5"
        >
          5
        </button>
        <button
          class="c5"
        >
          6
        </button>
        <button
          class="c5"
        >
          -
        </button>
        <button
          class="c5"
        >
          1
        </button>
        <button
          class="c5"
        >
          2
        </button>
        <button
          class="c5"
        >
          3
        </button>
        <button
          class="c5"
        >
          +
        </button>
        <button
          class="c5"
        >
          00
        </button>
        <button
          class="c5"
        >
          0
        </button>
        <button
          class="c5"
        >
          ,
        </button>
        <button
          class="c6"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
